_drafts/conviva-dag-process-time-prod-issue-draft.md

```diff
@@ -0,0 +1,69 @@
+---
+layout: post
+title: "The Problem With Concurrent Hashmaps"
+category: concurrency
+---
+
+## Setting The Stage
+
+We recently faced an interesting production issue at [Conviva](https://www.conviva.com/), related to the time taken by our compute engine to process customer events.
+
+On Feb 2nd, we noticed a recurring issue related to a specific customer where their P99 for processing time would absolutely shoot up. It was strange because we could only see it for a single customer which pointed to the issue most likely having to do with the customer's specific DAG.
+
+![](/assets/img/conviva/rtve/traffic-from-gateway.png)
```

Comment:
Be careful about sharing production charts.  We don't want to share names of customers. ;-)